### DISCUSSION TITLE
trying to add pipes to en_coreference_web_trf from other model and all tokens are NOUNs

### DISCUSSION THREAD

**Discussion by afriedman412** [explosion/spaCy#12644] [Help: Coding & Implementations] [answered]
I'm working something that I uses coreferencing and the basic spacy pipes. I was using `en_coreference_web_trf` and `en_core_web_sm` separately and parsing the text twice, but that seemed inefficient. I figured I would combine the two.

It looks like this:
```
nlp = spacy.load("en_coreference_web_trf")
for p in ['lemmatizer', 'attribute_ruler', 'parser', 'tagger']:
    try:
        nlp.add_pipe(p, source=spacy.load("en_core_web_sm"), first=True)
    except ValueError:
        continue
```

But when I run `nlp(t)` (where `t` is some text), it tags every single token as a `NOUN`.

help?

**Comment by danieldk** [comment, marked as answer]
The `lemmatizer`, `parser` and `tagger` pipe use the `tok2vec` pipe to get the contextualized word representations. So you have to add the `tok2vec` pipe as well and ensure that it runs before the other pipes. E.g. the following does work correctly:

```python
nlp = spacy.load("en_coreference_web_trf")
for p in ['tok2vec', 'tagger', 'parser', 'lemmatizer', 'attribute_ruler']:
    try:
        nlp.add_pipe(p, source=spacy.load("en_core_web_sm"))
    except ValueError:
        continue
```

**Comment by afriedman412** [comment]
lol so I was just forgetting one of the pipes.

thanks -- it works now!!
